lib/hanami/model/configurator.rb

```diff
@@ -25,12 +25,22 @@ class Configurator
       # @api private
       attr_reader :_schema
 
+      # @since x.x.x
+      # @api private
+      attr_reader :callbacks
+
       # @since 0.7.0
       # @api private
       def self.build(&block)
         new.tap { |config| config.instance_eval(&block) }
       end
 
+      # @since x.x.x
+      # @api private
+      def initialize
+        @callbacks = Hash[before: [], after: []]
```

Comment:
Maybe we could add them if needed later on, unless someone comes with a use case.

For the names I do agree with @pascalbetz , before and after are very generic but not that much descriptive.